plugins/permissions/src/channels/deleteChannel.ts

```diff
@@ -15,10 +15,21 @@ export function deleteChannel(bot: BotWithCache) {
 
       if (guild.publicUpdatesChannelId === channelId) throw new Error("UPDATES_CHANNEL_CANNOT_BE_DELETED");
 
-      const isThread = [ChannelTypes.AnnouncementThread, ChannelTypes.PublicThread, ChannelTypes.PrivateThread]
-        .includes(channel.type);
-
-      requireBotGuildPermissions(bot, guild, isThread ? ["MANAGE_THREADS"] : ["MANAGE_CHANNELS"]);
+      requireBotGuildPermissions(
```

Comment:
i think this should all be written differently.

something like

const perms = [];
if (isThread) perms.push("managethreads");
else {
  perms.push(managechannels, "viewchannel");
  if (isVoice) perms.push("conect")
}

requirePermCheck(perms)